service-account/bound-sa-token.md

```diff
@@ -1,122 +0,0 @@
-%title: Bound Service Account Token
-%author: Krzysztof Ostrowski
-%date: 10.03.2023
-
--> # Bound Service Account Token <-
-
---------------------------------------------------
--> # Table of Contents <-
-==============
-
--> # What is a Service Account Token  <-
--> # Service Account Token as a YAML <-
--> # Old Service Account Token Behavior  <-
--> # What is a Bound Service Account Token  <-
--> # Lord of Tokens: Kubelet <-
--> # Lord of ServiceAccounts: Service Account Admission Controller <-
--> # Bound Service Account Token in use by a Pod <-
-
--------------------------------------------------
--> # What is a Service Account Token  <-
-
-An **identity** that an instance or an application can use to make **API requests**:
-
-1. Intended to be used for **processes**.
-2. Bound to a **namespace**.
-3. Being associated with a ServiceAccount provides a **Bearer Token**.
-
--------------------------------------------------
--> # Service Account Token as a YAML <-
-
-```yaml
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: my-sa
-  namespace: my-namespace
-```
-
--------------------------------------------------
--> # Old Service Account Token Behavior  <-
-
-
-1. Previously **didn't expire**.
-2. Previously could be **used anywhere**.
-3. Token was **stored in a secret**.
-
--------------------------------------------------
--> # What is a Bound Service Account Token  <-
-
-- A Service Account Token that is **bound** to an **audience and lifespan**.
-- Given by **volume projection** or **TokenRequestAPI**.
-
--------------------------------------------------
--> # Lord of Tokens: Kubelet <-
-
-- Tries to **rotate** it after 80% of its lifetime is over.
```

Comment:
`rotates the tokens after 80% of their lifetime`

Natural questions from the statement:
- what is the lifetime of the token?
- what does it mean kubelet rotates the tokens? Where does it rotate them?